src/App.tsx

```diff
@@ -1,14 +1,40 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoModal } from './components/TodoModal';
 
 export const App: React.FC = () => {
+  const [loading, setLoading] = useState(false);
+  const [loadingModal, setLoadingModal] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>(todos);
+  const [showModalBoolean, setShowModalBoolean] = useState(false);
+  const [todo, setTodo] = useState({
+    title: '',
+    id: 0,
+    completed: false,
+    userId: 0,
+  });
+
+  useEffect(() => {
+    const getTodosFromServer = async () => {
+      setLoading(true);
+      const result = await getTodos();
+
+      setTodos(result);
+      setVisibleTodos(result);
+      setLoading(false);
+    };
```

Comment:
Take out from useEffect

Reply:
I didn't take it out because there will be too many re-renders

Reply:
[useCallback](https://www.w3schools.com/react/react_usecallback.asp) it will be useful 
 If you have any questions, please ask them in fe_chat and tag mentors. It will be faster, because your answers in GitHub will be visible to mentor only after you re-request review